### ISSUE TITLE
global_step是否应放在if-block外更合适？

### ISSUE THREAD

**Issue by illusions-LYY** [lonePatient/albert_pytorch#50]
https://github.com/lonePatient/albert_pytorch/blob/e9dbe3ce9aa49e787774b050cbdc496046e0c5bf/run_classifier.py#L114

`            if (step + 1) % args.gradient_accumulation_steps == 0:
                optimizer.step()
                scheduler.step()  # Update learning rate schedule
                model.zero_grad()
                global_step += 1

            if args.local_rank in [-1, 0] and args.logging_steps > 0 and global_step % args.logging_steps == 0:
                #Log metrics
                if args.local_rank == -1:  # Only evaluate when single GPU otherwise metrics may not average well
                    evaluate(args, model, tokenizer)`

以上是run_classifier.py line110~119的代码。假如`args.gradient_accumulation_steps`取默认值1，则不会有任何问题；然而当设置`args.gradient_accumulation_steps`为其他值时，以`4`为例，外循环的前3步（即step=0~3）就无法通过line110的if判断，从而导致global_step一直为0，然后导致line116的if判断基本总能通过（缘由`global_step=0`时，`global_step % args.logging_steps == 0`恒成立），最终导致还没开始梯度更新，就做了3次无谓的`evaluate`。
所以这里可能存在一些瑕疵，不知为何当初选择将`global_step += 1`放在if-block之内呢？
